### DISCUSSION TITLE
No such table in custom create trigger

### DISCUSSION THREAD

**Discussion by frankvollebregt** [simolus3/drift#1616] [Q&A] [answered]
I wanted to add some triggers to my data, to help me sync the client and server data. To this end I want to (let's keep it simple for now for the purposes of the example) create a trigger on two of my tables.

Since I don't want to manually add the triggers by copy-pasting the same one multiple times, I opted to use the `Migrator`s `onCreate`. In there, I defined roughly the following:

```dart
String statement = "CREATE TRIGGER IF NOT EXISTS test_trigger AFTER INSERT ON some_table WHEN old <> new BEGIN ... END;";
await this.customStatement(statement);
```
_EDIT: It looks like the OLD <> NEW part won't work anyways, but the error is still present regardless_
Where I intend to use a `for` loop to iterate over my tables, creating the trigger for each one. Between `BEGIN` and `END` I insert an entry into another table (which keeps track of a 'queue' of unsynced local inserts).

When I run the app, I receive the following error:
```dart
[ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: SqliteException(1): no such table: main.some_table 
E/flutter ( 4192):   Causing statement: CREATE TRIGGER IF NOT EXISTS test_trigger ...
```
~The same error occurs when I run the same statement from some point within my app, where the database already exists and has an active connection and data in it.~ I also tried changing the table to use, but the same thing occurs.

I noticed that the schema name `main` is chosen as a default, and figured that might be different. I tried some different things like `db`, `appdatabase` and the filename of my local database file, but all returned the same exception:
```dart
Unhandled Exception: SqliteException(1): trigger test_trigger cannot reference objects in database [name I chose]
```

Can anyone point me in the right direction as to what might be going wrong here for me?

TL;DR: When creating a trigger in the migrator's onCreate, my tables cannot be found

**Comment by frankvollebregt** [comment, marked as answer]
As it turns out, by default `onCreate` returns `m.createAll()`, so when overriding/replacing/extending the `onCreate`, I should call this myself too.

https://drift.simonbinder.eu/docs/advanced-features/migrations/
